store/sqlstore/post_persistent_notification_store.go

```diff
@@ -0,0 +1,216 @@
+	count := len(postIds)
+	if count == 0 {
+		return nil
+	}
+
+	builder := s.getQueryBuilder().
+		Update("PersistentNotifications").
+		Set("DeleteAt", model.GetMillis()).
+		Where(sq.Eq{"PostId": postIds})
+
+	_, err := s.GetMasterX().ExecBuilder(builder)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete notifications for posts %s", postIds)
+	}
+
+	return nil
+}
+
+func (s *SqlPostPersistentNotificationStore) UpdateLastActivity(postIds []string) error {
+	count := len(postIds)
+	if count == 0 {
+		return nil
+	}
+
+	builder := s.getQueryBuilder().
+		Update("PersistentNotifications").
+		Set("LastSentAt", model.GetMillis()).
+		Set("SentCount", sq.Expr("SentCount+1")).
+		Where(sq.Eq{"PostId": postIds})
+
+	_, err := s.GetMasterX().ExecBuilder(builder)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update lastSentAt for posts %s", postIds)
+	}
+
+	return nil
+}
+
+func (s *SqlPostPersistentNotificationStore) DeleteByChannel(channelIds []string) error {
+	count := len(channelIds)
+	if count == 0 {
+		return nil
+	}
+
+	deleteAt := model.GetMillis()
+	var builder sq.UpdateBuilder
+	builderType := s.getQueryBuilder()
+	if s.DriverName() == model.DatabaseDriverMysql {
+		builder = builderType.
+			Update("PersistentNotifications, Posts").
+			Set("PersistentNotifications.DeleteAt", deleteAt)
+	}
+
+	if s.DriverName() == model.DatabaseDriverPostgres {
+		builder = builderType.
+			Update("PersistentNotifications").
+			Set("DeleteAt", deleteAt).
+			From("Posts")
+	}
+
+	builder = builder.Where(sq.And{
+		sq.Expr("Id = PostId"),
+		sq.Eq{"ChannelId": channelIds},
+	})
+
+	_, err := s.GetMasterX().ExecBuilder(builder)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete notifications for channels %s", channelIds)
+	}
+
+	return nil
+}
+
+func (s *SqlPostPersistentNotificationStore) DeleteByTeam(teamIds []string) error {
+	count := len(teamIds)
+	if count == 0 {
+		return nil
+	}
+
+	deleteAt := model.GetMillis()
+	var builder sq.UpdateBuilder
+	builderType := s.getQueryBuilder()
+	if s.DriverName() == model.DatabaseDriverMysql {
+		builder = builderType.
+			Update("PersistentNotifications, Posts, Channels").
+			Set("PersistentNotifications.DeleteAt", deleteAt)
+	}
+
+	if s.DriverName() == model.DatabaseDriverPostgres {
+		builder = builderType.
+			Update("PersistentNotifications").
+			Set("DeleteAt", deleteAt).
+			From("Posts, Channels")
+	}
+
+	builder = builder.Where(sq.And{
+		sq.Expr("Posts.Id = PersistentNotifications.PostId"),
+		sq.Expr("Posts.ChannelId = Channels.Id"),
+		sq.Eq{"Channels.TeamId": teamIds},
+	})
+
+	_, err := s.GetMasterX().ExecBuilder(builder)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete notifications for teams %s", teamIds)
+	}
+
+	return nil
+}
+
+func getPersistentNotificationsPaginationBuilder(builder sq.SelectBuilder, pagination model.CursorPagination) sq.SelectBuilder {
+	var sort string
+	if pagination.Direction != "" {
```

Comment:
This `if` has no effect (maybe a micro-optimization on checking the value, a thing that I'm not sure about), and it reduces the readability of this code. I would get rid of that.